homeassistant/components/saj/sensor.py

```diff
@@ -178,15 +101,15 @@ def __init__(self, inverter: SAJInverter, pysaj_sensor: pysaj.Sensor) -> None:
     @property
     def name(self):
         """Return the name of the sensor."""
-        if self._inverter.name != DEFAULT_NAME:
-            return f"{self._inverter.name} {self._sensor.name}"
+        if self.coordinator.name != DEFAULT_NAME:
+            return f"{self.coordinator.name} {self._sensor.name}"
 
         return f"SAJ Inverter {self._sensor.name}"
 
     @property
     def state(self):
         """Return the state of the sensor."""
-        if not self._inverter.available:
+        if not self.coordinator.last_update_success:
```

Comment:
When using the coordinator, this check isn't needed.
`state()` isn't called when the entity isn't unavailable.

Reply:
some entity like "today_yield" is still available after sundown when the inverter is not reachable (assumed state?)
pls see comment in lines below

Reply:
@frenck ping